### PR TITLE
MEN-5720: Acceptance tests for webhooks functionality

### PR DESCRIPTION
**Pull request by alfrunes**

### CODE REVIEW

**Comment by alfrunes** [comment]
@mender-test-bot start pipeline --fast

**Review by tranchitella — APPROVED** [review]

File: tests/mmock/config/test_sync_aws_iotcore/iotcore_describe_thing_not_found_9b2083e4-83b6-41d3-b089-4d2ec137620b [hunk 1]
```diff
@@ -0,0 +1,5 @@
+control:
+  scenario:
+    name: class
+    requiredState:
+      - TestSyncAWSIoTCore
```
[the remark this answers is not in the document]
Comment (alfrunes) [reply tests/mmock/config/test_sync_aws_iotcore/iotcore_describe_thing_not_found_9b2083e4-83b6-41d3-b089-4d2ec137620b:5]: Oops... My snippet appending this trailer to all the yaml files did not quote the filename shell variable.

File: docs/internal_api.yml [hunk 2]
```diff
@@ -199,12 +199,57 @@ components:
...
+      description: >-
+        AuthSet describes the identity a device use to authenticate with the
+        Mender servier.
+      properties:
+        id:
+          type: string
+          description: The unique ID of the authentication set.
+        device_id:
+          type: string
+          description: The unique ID of the device the authentication set belongs.
```
[the remark this answers is not in the document]
Comment (alfrunes) [reply docs/internal_api.yml:237]: This is just a copy of the deviceauth model of the device. I didn't want to alter it.

File: docs/internal_api.yml [hunk 3]
```diff
@@ -199,12 +199,57 @@ components:
...
-        device_id:
+        id:
           type: string
           format: uuid
           description: ID of the new device.
+        status:
+          type: string
+          enum:
+            - accepted
+            - preauth
```
[the remark this answers is not in the document]
Comment (alfrunes) [reply docs/internal_api.yml]: Corrected it to `preauthorized`.

File: tests/tests/utils.py [hunk 4]
```diff
@@ -22,17 +22,25 @@
...
+                compare_expectations(expected[i], actual[i])
+        elif isinstance(expected, dict):
+            assert isinstance(actual, dict)
+            for key, expected_value in expected.items():
+                assert key in actual
+                compare_expectations(expected_value, actual[key])
         else:
-            assert expected_value == actual_value, f"{expected} != {actual}"
+            assert expected == actual, f"{expected} != {actual}"
+    return True
```
[the remark this answers is not in the document]
Comment (alfrunes) [reply tests/tests/utils.py:43]: It's just so that you can call it in an assertion statement.

**Comment by alfrunes** [comment]
@mender-test-bot start pipeline --fast

**Review by tranchitella — APPROVED** [review]

**Pull request merged**